lib/spack/spack/util/windows_registry.py

```diff
@@ -280,25 +346,11 @@ def _traverse_subkeys(self, stop_condition, collect_all_matching=False):
                         collection.append(key)
                     else:
                         return key
-                queue.extend(key.subkeys)
+                if depth_search:
+                    queue.extend(key.subkeys)
             return collection if collection else None
 
-    def _find_subkey_s(self, search_key, collect_all_matching=False):
-        """Retrieve one or more keys regex matching `search_key`.
-        One key will be returned unless `collect_all_matching` is enabled,
-        in which case call matches are returned.
-
-        Args:
-            search_key (str): regex string represeting a subkey name structure
-                              to be matched against.
-                              Cannot be provided alongside `direct_subkey`
-            collect_all_matching (bool): No-op if `direct_subkey` is specified
-        Return:
-            the desired subkey as a RegistryKey object, or none
-        """
-        return self._traverse_subkeys(search_key, collect_all_matching=collect_all_matching)
-
-    def find_subkey(self, subkey_name):
+    def find_subkey(self, subkey_name, depth=True):
```

Comment:
Update docstring with new arg. If I understand correctly, this determines whether you list the first level or whether you recurse into subkeys. Typically we have used "recursive" for this notion, which I think is also clearer in this context.

Reply:
Sounds good, amended.